### ISSUE TITLE
B012 false positive when break is in a loop in a finally block

### ISSUE THREAD

**Issue by davidism** [PyCQA/flake8-bugbear#106]
#99 and #100 added a new check B012 warning about `return`/`continue`/`break` inside `finally` blocks. The implementation raises a false positive when the `continue` or `break` is in a loop in the `finally` block.

Here's some [code in Werkzeug](https://github.com/pallets/werkzeug/blob/ea0d6b95fb118f33025163b7441577ee5755d971/src/werkzeug/formparser.py#L126-L142) demonstrating the issue.

```python
def exhaust_stream(f):
    """Helper decorator for methods that exhausts the stream on return."""

    def wrapper(self, stream, *args, **kwargs):
        try:
            return f(self, stream, *args, **kwargs)
        finally:
            exhaust = getattr(stream, "exhaust", None)
            if exhaust is not None:
                exhaust()
            else:
                while 1:
                    chunk = stream.read(1024 * 64)
                    if not chunk:
                        break

    return update_wrapper(wrapper, f)
```

Perhaps the code itself could be written differently regardless, I haven't gone down that path yet. But it doesn't seem like the code would cause exceptions to be silenced, which is what B012 is warning about.

cc @joaoe

**Comment by cooperlees** [comment]
Thanks for the great report. We’ll look in to it. I’m not sure if we want to detect if it’s in a loop or not.

**Comment by joaoe** [comment]
Ah, nice catch. I'll gladly fix it.

**Comment by cooperlees** [comment]
Will close when I push a new version to PyPI.

**Comment by cooperlees** [comment]
20.1.1 should have this fixed. Please re-open if not.

**Comment by davidism** [comment]
Seems to be working, thanks. Working so well, in fact, that I can't get the original error to trigger after downgrading flake8. I'm sure it was happening though! :man_shrugging:
